phpdotnet/phd/Package/PHP/XHTML.php

```diff
@@ -523,29 +510,109 @@ public function format_methodsynopsis($open, $name, $attrs, $props) {
+                }
+            }
+        }
+
+        $type = implode($this->type_separator ?? '', $types);
+        if (count($types) > 1) {
+            $type = '<span class="type">' . $type . '</span>';
+        }
+        $this->simple_nullable = null;
+
+        return $type;
+    }
+
     public function format_type_methodsynopsis_text($type, $tagname) {
         $this->cchunk["methodsynopsis"]["returntypes"][] = $type;
 
         return "";
     }
 
-    public function format_type_if_object_or_pseudo_text($type, $tagname) {
-        if (strtolower($type) === "null" && $this->simple_nullable) {
-            return "";
+    public function format_methodparam_parameter($open, $name, $attrs, $props) {
+        $type = "";
+        if ($open) {
+            if ($this->cchunk["methodparam"]["paramtypes"]) {
+                $type = $this->format_types(
+                    $this->cchunk["methodparam"]["type_separator"],
+                    $this->cchunk["methodparam"]["paramtypes"]
+                );
+            }
+            $this->cchunk["methodparam"]["type_separator"] = "";
+            $this->cchunk["methodparam"]["paramtypes"] = [];
+        }
+        return $type . parent::format_methodparam_parameter($open, $name, $attrs, $props);
+    }
+
+    public function format_methodparam($open, $name, $attrs) {
+        if ($open) {
+            $this->cchunk["methodparam"]["paramtypes"] = [];
+            $this->cchunk["methodparam"]["type_separator"] = "";
+            return parent::format_methodparam($open, $name, $attrs);
         }
 
-        if (in_array(strtolower($type), array("bool", "int", "double", "boolean", "integer", "float", "string", "array", "object", "resource", "null"))) {
-            return false;
+        if ($this->params["opt"] && !$this->params["init"]) {
+            return '<span class="initializer"> = ?</span></span>';
         }
+        $this->params["init"] = false;
+        $this->cchunk["methodparam"]["paramtypes"] = [];
+        $this->cchunk["methodparam"]["type_separator"] = "";
+        return "</span>";
+    }
 
-        return self::format_type_text($type, $tagname);
+    public function format_type_methodparam($open, $tag, $attrs) {
+        if ($open) {
+            if (isset($attrs[Reader::XMLNS_DOCBOOK]["class"])) {
+                if ($attrs[Reader::XMLNS_DOCBOOK]["class"] === "union") {
+                    $this->cchunk["methodparam"]["type_separator"] = "|";
+                } else {
+                    $this->cchunk["methodparam"]["type_separator"] = "&";
+                }
+            }
+            $this->cchunk["methodparam"]["paramtypes"] = [];
+        }
+        return "";
+    }
+
+    public function format_type_methodparam_text($type, $tagname) {
+        $this->cchunk["methodparam"]["paramtypes"][] = $type;
+
+        return "";
     }
 
     public function format_type_text($type, $tagname) {
+        $type = trim($type);
```

Comment:
I'm not super sure we should do this, maybe add a warning? Because the sources shouldn't be having a whitespace ideally inside a `<type>` tag.

Reply:
See my longer comment on warnings above. :-)

Reply:
I'll remove this `trim()` though and the test case from the test files.